moveit_ros/planning/planning_scene_monitor/src/current_state_monitor.cpp

```diff
@@ -68,29 +70,29 @@ planning_scene_monitor::CurrentStateMonitor::~CurrentStateMonitor()
 
 robot_state::RobotStatePtr planning_scene_monitor::CurrentStateMonitor::getCurrentState() const
 {
-  boost::mutex::scoped_lock slock(state_update_lock_);
+  std::unique_lock<std::mutex> slock(state_update_lock_);
```

Comment:
Is there a reason we are not using scoped_lock?

Reply:
I think because `scoped_lock` get added in C++17 standard [see](https://en.cppreference.com/w/cpp/thread/scoped_lock)

Reply:
I used unqiue_lock because it turn out that in the boost implementation it's just a typedef to unique_lock [see](https://stackoverflow.com/a/9494139)